SerialPrograms/Source/PokemonHome/Programs/PokemonHome_BoxSorting.cpp

```diff
@@ -435,81 +465,85 @@ void BoxSorting::program(SingleSwitchProgramEnvironment& env, BotBaseContext& co
 
-    //this need to be separated into functions when I will redo the whole thing but I just wanted it to work
-
-    //going thru the sorted list one by one and for each one go through the current pokemon layout to find the closest possible match to fill the slot
-    for (int box_nb = 0; box_nb < int(box_sorted.size()); ++box_nb) {
-        for (int column = 0; column < 5; ++column) {
-            for (int line = 0; line < 6; ++line) {
-                bool found_pkmn = false;
-                for (int box_nb_s = 0; box_nb_s < int(box_data.size()); ++box_nb_s) {
-                    for (int column_s = 0; column_s < 5; ++column_s) {
-                        for (int line_s = 0; line_s < 6; ++line_s) {
-                            //check for a match and also check if the pokemon is not already in the slot
-                            if(box_sorted[box_nb][column][line] == box_data[box_nb_s][column_s][line_s] && box_data[box_nb_s][column_s][line_s] != box_data[box_nb][column][line]){
-
-                                //moving cursor to the pokemon to pick it up
-                                ss << "Moving cursor to " << box_nb_s <<" "<< column_s <<" "<< line_s;
-                                env.console.log(ss.str());
-                                ss.str("");
-                                move_cursor_to(env, context, box_nb_s, column_s, line_s, &cur_cursor, GAME_DELAY);
-                                pbf_press_button(context, BUTTON_Y, 10, GAME_DELAY+30);
-                                //moving to destination to place it or swap it
-                                ss << "Moving "<< box_data[box_nb_s][column_s][line_s] <<" to " << box_nb <<" "<< column <<" "<< line;
-                                env.console.log(ss.str());
-                                ss.str("");
-                                move_cursor_to(env, context, box_nb, column, line, &cur_cursor, GAME_DELAY);
-                                pbf_press_button(context, BUTTON_Y, 10, GAME_DELAY+30);
-                                context.wait_for_all_requests();
-                                box_data[box_nb_s][column_s][line_s] = box_data[box_nb][column][line]; //update the list in case of a pokemon swap
-                                box_data[box_nb][column][line] = std::nullopt; //delete pokemon data to make sure it does not match again
-                                found_pkmn = true;
-                                break;
-                            }
-                        }
-                        if(found_pkmn){
-                            break;
-                        }
-                    }
-                    if(found_pkmn){
-                        break;
-                    }
-                }
+    // going thru the sorted list one by one and for each one go through the current pokemon layout to find the closest possible match to fill the slot
+    for (int poke_nb_s = 0; poke_nb_s < int(boxes_sorted.size()); poke_nb_s++) {
```

Comment:
poke_nb_s could be a size_t instead to avoid the conversion

Reply:
I didn't do that because poke_nb and poke_nb_s are used as int in the body of the for loop